sklearn/calibration.py

```diff
@@ -658,3 +660,232 @@ def calibration_curve(y_true, y_prob, *, normalize=False, n_bins=5,
     prob_pred = bin_sums[nonzero] / bin_total[nonzero]
 
     return prob_true, prob_pred
+
+
+class CalibrationDisplay:
+    """Calibration visualization.
+
+    It is recommend to use :func:`~sklearn.calibration.plot_calibration_curve`
+    to create a visualizer. All parameters are stored as attributes.
+
+    Read more in the :ref:`User Guide <calibration>`.
+
+    Parameters
+    -----------
+    prob_true : ndarray
+        The proportion of samples whose class is the positive class (fraction
+        of positives), in each bin.
+
+    prob_pred : ndarray
+        The mean predicted probability in each bin.
+
+    y_prob : ndarray of shape (n_samples,)
+        Probability estimates for the positive class, for each sample.
+
+    brier_value : int or None
+        The Brier score value. If `None`, the Brier score is not shown.
```

Comment:
Since Brier score is not necessarily the best way to quantify classifiers calibration (as  discussed in #10883),  and https://github.com/scikit-learn/scikit-learn/pull/11096 will probably be a better default metric for this plot.

In the mean time I think we should make remove Brier score from the `CalibrationDisplay` class and instead add the possibility to pass a custom label value that would be used in the examples to keep the current display of the Brier score while waiting for a better alternative metric.

Reply:
I feel like we should just scrap the brier score completely, from `plot_calibration_curve()` and the examples, and I would just add them once #11096 is merged? We already said that brier score is not being interpreted correctly, so why use it in the examples?

Alternatively I can wait until #11096 is merged before I finish this PR?

Reply:
Ok for scrapping for now.